### DISCUSSION TITLE
Fixing "fail: No bucket named" when bootstrapping using --bootstrap-bucket-name

### DISCUSSION THREAD

**Discussion by reikje** [aws/aws-cdk#21051] [General]
I have used the `--bootstrap-bucket-name` flag when running `cdk bootstrap`. Now `cdk deploy` fails because it cannot find the right bucket. 

> [100%] fail: No bucket named 'xyz-123456789-us-east-1'. Is account 123456789 bootstrapped?

I understand that I should provide a custom [stack synthesizer](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html#bootstrapping-synthesizers) but what is the property I should set? Can I use `cdk.json` instead, as I have already set a custom qualifier there - and if I can, what is the option I should set in that file?

**Comment by nagkumar** [comment]
+1 same issue for me.. none of CDK examples are working.. may be due to security policy issue, no CDK developer guide or anywhere it is told.. pulumi was a cakewalk.. cdk seem  to be too painful

**Comment by indrora** [comment]
You'll need to adjust your app stack slightly.

```typescript
// where your stack is set up:
new MyStack(this, 'MyStack', {
  synthesizer: new DefaultStackSynthesizer({
    fileAssetsBucketName: 'MySuperSpecialBoostrapBucket',
    bucketPrefix: '',
  }),
});
```

Details are [here](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html#bootstrapping-custom-synth) under the heading "Changing the resource names".

**Comment by silver-mx** [comment]
I solved my issue by deleting the `CDKToolkit` stack created in Cloudformation. The stack existed and was in an inconsistent state. So this was the solution for me:

`aws cloudformation delete-stack --stack-name CDKToolkit --profile my-profile`

and then bootstrap again:

`cdk bootstrap aws://ACCOUNT/REGION --profile my-profile`

Source: https://stackoverflow.com/a/71283964/677096

**Reply by orimslala** [reply]
Thank you! worked for me

**Reply by VishalVinayRam** [reply]
Thank you , it worked for me

**Reply by shashiks** [reply]
This just works! Thanks

**Reply by menoudyann** [reply]
Worked for me, thank you !

**Reply by zakdma** [reply]
Helped
Thank you!

**Reply by androyt1** [reply]
Thank you
